### ISSUE TITLE
Failed to load DLL - The specified procedure was not found

### ISSUE THREAD

**Issue by ghuDaYuYu** [ghDaYuYu/foo_discogs#91]
Not a crash, discogger just won't load, I guess?

![11-02-2023-234042-2](https://user-images.githubusercontent.com/13855295/218286899-09f86972-ff7e-4196-a16c-976809aee537.png)

It appears every time i start the player.

> Failed to load DLL: foo_discogger.dll  
> Reason: The specified procedure was not found.  

Reinstalling, upgrading, rolling back to an older version of the plugin does not help.

foobar2000 v2.0 beta 23 [x86]

After checking, the same error occurs with clean foobar2000 v1.6.16

I definitely used one of the old versions of discogger with foobar v2.x, I don't remember which one. But everything worked up to some point.

Does the plugin have any system requirements for the operating system? Could they have changed?

Windows 10 x64 1607.

_Originally posted by @Lichtenshtein in https://github.com/ghDaYuYu/foo_discogs/issues/78#issuecomment-1426908690_

**Comment by ghuDaYuYu** [comment]
The plugin is reported to be running under Wine/Linux. I test it occasionally in that platform as well. I mention it as Wine would surely complain if there were special requirements to run it.
System requirements did not change, I believe, as the message suggests, there is a problem calling an internal procedure, maybe some incompatibility with foobar2000 itself.

Have you tried using a clean install other than v1.6.16 ?

**Comment by Lichtenshtein** [comment]
foobar2000 v1.6.10 portable installation and foo_discogger 1.0.15.1 - error
foobar2000 v1.6.10 standart installation and foo_discogger 1.0.15.1 - error
foobar2000 v1.6.4 portable installation and foo_discogger 1.0.15.1  - error
foobar2000 v1.6.10 portable installation and foo_discogger 1.0.16.1 - error
foobar2000 v1.6.10 portable installation and foo_discogger 1.0.15.1 - error
foobar2000 v1.6.10 portable installation and foo_discogger 1.0.8 - no error starts fine
foobar2000 v1.6.10 portable installation and foo_discogger 1.0.9 - no error starts fine
foobar2000 v1.6.10 portable installation and foo_discogger 1.0.10 - no error starts fine
foobar2000 v1.6.10 portable installation and foo_discogger 1.0.11 - no error starts fine
foobar2000 v1.6.10 portable installation and foo_discogger 1.0.11.1 - no error starts fine
foobar2000 v1.6.10 portable installation and foo_discogger 1.0.12 - no error starts fine
foobar2000 v1.6.10 portable installation and foo_discogger 1.0.12.2 - no error starts fine
foobar2000 v1.6.10 portable installation and foo_discogger 1.0.13 - **error**
foobar2000 v2.0 beta 24 portable installation and foo_discogger 1.0.13 - error
foobar2000 v2.0 beta 24 standart installation and foo_discogger 1.0.13 - error
foobar2000 v2.0 x64 beta 24 standart installation and foo_discogger 1.0.13 - error
foobar2000 v2.0 x64 beta 24 portable installation and foo_discogger 1.0.13 - error

**Comment by ghuDaYuYu** [comment]
All those are clean portable installations on the same machine/hard drive  or different machines ?
Does not seem random at first sight... what about foobar2000 v2.0 x64 beta 24 and [v1.0.16.1](https://github.com/ghDaYuYu/foo_discogs/releases/tag/v1.0.16.1) or [v1.0.15.1](https://github.com/ghDaYuYu/foo_discogs/releases/tag/v1.0.15.1)?

**Comment by Lichtenshtein** [comment]
> All those are clean portable installations on the same machine/hard drive or different machines ?

Yes, clean portable and standart installations on same machine using Sandboxie v5.60.3
https://github.com/sandboxie-plus/Sandboxie

foobar2000 v2.0 x64 beta 24 portable installation and foo_discogger 1.0.15.1 - error
foobar2000 v2.0 x64 beta 24 standart installation and foo_discogger 1.0.15.1 - error
foobar2000 v2.0 x64 beta 24 portable installation and foo_discogger 1.0.16.1 - error
foobar2000 v2.0 x64 beta 24 standart installation and foo_discogger 1.0.16.1 - error

In my opinion, the error is due to some changes made since the version 1.0.13 of the plugin.

**Comment by ghuDaYuYu** [comment]
Might be... but project file did not change between 12.2 and 13.
Not sure, to tell the truth. Could it have some relation to the system %PATH% ? I can only suspect something wrong with the .dll versions being loaded.

**Comment by ghuDaYuYu** [comment]
Sorry for not being of much help, but I would need to install and test from Sandboxie myself.
I will found it strange that foo_discogger is the only component having trouble to load in your configuration.
I've just encounter this error, once. But it had a factor of apparent randomness.
Let us know if you either find a solution or get to a conclusion.

**Comment by ghuDaYuYu** [comment]
Still looking into this, there are more people reporting problems running  versions over 12.2 in Windows 7.

**Comment by ghuDaYuYu** [comment]
There is a new release fixing an installation failure 'procedure was not found' in Windows 7.
It might also solve the issue in Sandboxie, would you give it a try?

**Comment by Lichtenshtein** [comment]
> would you give it a try?

YES, gladly! 

It's working!

Tested with versions in Sandboxie:
foobar2000 v1.6.16 standart installation and foo_discogger v1.0.16.4 - working
foobar2000 v2.0 beta 24 portable installation and foo_discogger v1.0.16.4 - working
foobar2000 x64 v2.0 beta 24 portable installation and foo_discogger v1.0.16.4 - working

And in my personal foobar2000 profile, it works too. Thank you!

For the sake of interest, can you briefly describe what the problem was with versions after 1.0.12.2?

**Comment by ghuDaYuYu** [comment]
What? Nice one!

It was out of my ignorance. Technically, after 12.2, I added a new option to the component (UI Options->Expanded content area font). Sadly, I called a theme-related function, OpenThemeDataForDpi, to implement it with catastrophic outcomes, as you have realised.

![DependencyWalker](https://user-images.githubusercontent.com/62167612/220000029-3fcf15f9-6fef-4a98-ad95-603da117a9af.PNG)

I am sorry that you had to go through all that testing... The component is available for Windows 7 (and apparently Windows 10 Sandboxies) thanks to you now!

Best regards
